internal/controllers/nmc_reconciler.go

```diff
@@ -923,12 +934,16 @@ var (
 	}
 )
 
-func (p *podManagerImpl) baseWorkerPod(
-	ctx context.Context,
-	nodeName string,
-	item *kmmv1beta1.ModuleItem,
-	owner client.Object,
-) (*v1.Pod, error) {
+type ContainerArg struct {
+	KmodsPath        string
+	KmodsHostPath    string
+	FirmwarePath     string
+	FirmwareHostPath string
+}
+
+func (p *podManagerImpl) baseWorkerPod(ctx context.Context, nmc client.Object, item *kmmv1beta1.ModuleItem,
```

Comment:
This is an internal interface used only by NMC reconciler. Both item and moduleConfig are part of NMS object, why not just path NMS and let baseWorkerPod find whaever info it needs?

Reply:
Because the 2 consumers are using this method with different input. One of them use a `spec` and another one the `status` so I had to go with the next larger object to contain everything I need.